source/security/firmware-update.rst

```diff
@@ -1,37 +1,372 @@
+       ///
+       UINT16 EmbeddedDriverCount;
+    
+       ///
+       /// The number of payload items included in the capsule and the number of
+       /// corresponding offsets stored in the ItemOffsetList array.
+       ///
+       UINT16 PayloadItemCount;
+    
+       ///
+       /// Variable length array of dimension [EmbeddedDriverCount + PayloadItemCount]
+       /// containing offsets of each of the drivers and payload items contained within the capsule
+       ///
+       // UINT64 ItemOffsetList[];
+     } EFI_FIRMWARE_MANAGEMENT_CAPSULE_HEADER;
+    
+      #define EFI_FIRMWARE_MANAGEMENT_CAPSULE_HEADER_INIT_VERSION       0x00000001
+
+EFI_FIRMWARE_MANAGEMENT_CAPSULE_HEADER is followed by one or multiple update images each preceeded by EFI_FIRMWARE_MANAGEMENT_CAPSULE_IMAGE_HEADER header and is defined in UEFI specification as follows 
+
+.. code-block:: c
+
+     typedef struct {
+       UINT32   Version;
+    
+       ///
+       /// Used to identify device firmware targeted by this update. This guid is matched by
+       /// system firmware against ImageTypeId field within a EFI_FIRMWARE_IMAGE_DESCRIPTOR
+       ///
+       EFI_GUID UpdateImageTypeId;
+    
+       ///
+       /// Passed as ImageIndex in call to EFI_FIRMWARE_MANAGEMENT_PROTOCOL.SetImage ()
+       ///
+       UINT8    UpdateImageIndex;
+    
+       UINT8    reserved_bytes[3];
+    
+       ///
+       /// Size of the binary update image which immediately follows this structure
+       ///
+       UINT32   UpdateImageSize;
+    
+       ///
+       /// Size of the VendorCode bytes which optionally immediately follow binary update image in the capsule
+       ///
+       UINT32   UpdateVendorCodeSize;
+    
+       ///
+       /// The HardwareInstance to target with this update. If value is zero it means match all
+       /// HardwareInstances. This field allows update software to target only a single device in
+       /// cases where there are more than one device with the same ImageTypeId GUID.
+       /// This header is outside the signed data of the Authentication Info structure and
+       /// therefore can be modified without changing the Auth data.
+       ///
+       UINT64   UpdateHardwareInstance;
+     } EFI_FIRMWARE_MANAGEMENT_CAPSULE_IMAGE_HEADER;
+
+     #define EFI_FIRMWARE_MANAGEMENT_CAPSULE_IMAGE_HEADER_INIT_VERSION 0x00000002
+
+.. note:: Please refer to UEFI specification for more details about capsule header and data.
 
 Triggering Firmware Update 
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Triggering firmware update is a platform specific mechanism. The location of the firmware update capsule image is passed to Firmware update payload through configuration data.
+Triggering firmware update is a platform specific mechanism.
+
+|SPN| uses platform dependent register that can survive a reset to signal firmware update. Please refer to **Triggering Firmware Update** section of desired board page in **Supported Hardware** to find Sample implementation.

```

Comment:
platform specific instead of platform dependent

Reply:
Done